orca-echo/src/main/java/com/netflix/spinnaker/orca/echo/util/ManualJudgmentAuthorization.java

```diff
@@ -0,0 +1,172 @@
+
+  private final Front50Service front50Service;
+  private final FiatPermissionEvaluator fiatPermissionEvaluator;
+
+  private final FiatStatus fiatStatus;
+  private final ObjectMapper objectMapper;
+
+  @Autowired
+  public ManualJudgmentAuthorization(
+      Optional<Front50Service> front50Service,
+      Optional<FiatPermissionEvaluator> fiatPermissionEvaluator,
+      FiatStatus fiatStatus,
+      ObjectMapper objectMapper) {
+    this.front50Service = front50Service.orElse(null);
+    this.fiatPermissionEvaluator = fiatPermissionEvaluator.orElse(null);
+
+    this.fiatStatus = fiatStatus;
+    this.objectMapper = objectMapper;
+  }
+
+  /**
+   * A manual judgment will be considered "authorized" if the current user has at least one of the
+   * required judgment roles _and_ read/write/execute permissions on the application being judged.
+   *
+   * @param application Application being judged
+   * @param requiredJudgmentRoles Required judgment roles
+   * @param currentUser User that has attempted this judgment
+   * @return whether or not {@param currentUser} has authorization to judge
+   */
+  public boolean isAuthorized(
+      String application, Collection<String> requiredJudgmentRoles, String currentUser) {
+    if (!fiatStatus.isEnabled() || requiredJudgmentRoles.isEmpty()) {
+      return true;
+    }
+
+    if (Strings.isNullOrEmpty(currentUser)) {
+      return false;
+    }
+
+    UserPermission.View permission = fiatPermissionEvaluator.getPermission(currentUser);
+    if (permission == null) { // Should never happen?
+      log.warn("Attempted to get user permission for '{}' but none were found.", currentUser);
+      return false;
+    }
+
+    return isAuthorized(
+        requiredJudgmentRoles,
+        permission.getRoles().stream().map(Role.View::getName).collect(Collectors.toList()),
+        getApplicationPermissions(application));
+  }
+
+  private boolean isAuthorized(
+      Collection<String> requiredJudgmentRoles,
+      Collection<String> currentUserRoles,
+      Map<String, Object> applicationPermissions) {
+    if (requiredJudgmentRoles == null || requiredJudgmentRoles.isEmpty()) {
+      return true;
+    }
+
+    return Sets.intersection(new HashSet<>(requiredJudgmentRoles), new HashSet<>(currentUserRoles))
+            .size()
+        > 0;
+
+    //    boolean isAuthorizedGroup = false;
+    //
+    //    for (String role : currentUserRoles) {
+    //      if (requiredJudgmentRoles.contains(role)) {
+    //        for (Map.Entry<String, Object> entry :
+    //            applicationPermissions.entrySet()) { // get the application permission roles.
+    //          if (Authorization.CREATE.name().equals(entry.getKey())
+    //              || Authorization.EXECUTE.name().equals(entry.getKey())
+    //              || Authorization.WRITE.name().equals(entry.getKey())) {
+    //            // If the application permission roles has 'CREATE, EXECUTE, WRITE', then user is
```

Comment:
Commented out this logic as I believe it's incorrect.

The comment indicates something about CREATE/EXECUTE/WRITE but is really just checking that one of the required judgment roles has CREATE, EXECUTE or WRITE access to the application.

I don't know what the point of this check is.

IMO, this would be useful feature if we just checked user roles against the required judgment roles. This would allow someone to restrict who is able to perform a judgment, which I thought was the original intent of this feature.